lib/cast.js

```diff
@@ -225,6 +225,13 @@ module.exports = function cast(schema, obj, options) {
             }
           }
         }
+      } else if (val.constructor.name === 'Array' && ['Buffer', 'Array'].indexOf(schematype.instance) === -1) {
+        var casted = [];
```

Comment:
My preference would have been to put this logic into the individual `cast` methods for each type, but the problem is that in those `cast` methods, I have no idea if the value is actually supposed to be an array of values.  e.g. I could have a schema with:
```js
{
  someKey: { type: [Number] }
}
```
OR, I could have a schema with:
```js
{
  someKey: { type: Number } 
}
```

However, in `lib/schema/number.js`'s `cast` method, I don't know if the value is _supposed to be an array_, or if the value can simply be one of any number of values and thus can be converted to `$in` query.  

I would have passed another variable into the cast method at line `236` to contain the `schematype`, but the problem is having more than 1 argument will change how the method behaves entirely.

Long story short, I'm happy to hear possible improvements or a better location for this logic that can still pass the tests.